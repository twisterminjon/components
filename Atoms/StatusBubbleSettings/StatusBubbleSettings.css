.statusbubblesettings {
  display: flex;
  justify-content: center;
  align-items: center;

  height: 15px;
  width: 15px;

  border-radius: 11px;
  border: 1px solid var(--white);

  color: var(--white);
  background-color: var(--statusbubble_offline);
}

.statusbubble--available {
  background-color: var(--statusbubble_available);
}

.statusbubble--busy {
  background-color: var(--statusbubble_busy);
}

.statusbubble--away {
  background-color: var(--statusbubble_away);
}

.statusbubble--offline {
  background-color: var(--statusbubble_offline);
}
