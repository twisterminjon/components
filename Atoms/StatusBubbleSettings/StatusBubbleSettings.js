import React from 'react'
import PropTypes from 'prop-types'

import IconHamburger from '../../Atoms/Icons/IconHamburger'
import { USER_STATUS_LIST } from '../../../constants'

import './StatusBubbleSettings.css'

StatusBubbleSettings.propTypes = {
  /** The status to to display in the bubble */
  status: PropTypes.oneOf(USER_STATUS_LIST).isRequired,

  /** ClassName for the wrapper */
  className: PropTypes.string,

  /** Style for the wrapper */
  style: PropTypes.object,
}

StatusBubbleSettings.defaultProps = {
  className: '',
  style: {},
}

export default function StatusBubbleSettings({ status, className, style }) {
  const colorClass = `statusbubble--${status}`

  return (
    <div className={`statusbubblesettings ${colorClass} ${className}`.trim()} style={style}>
      <IconHamburger size={9} color="var(--white)" />
    </div>
  )
}
