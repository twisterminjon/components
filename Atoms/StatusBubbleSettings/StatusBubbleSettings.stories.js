import React from 'react'

import { storiesOf } from '@storybook/react'
import centered from '@storybook/addon-centered/react'
import { withKnobs, select } from '@storybook/addon-knobs'

import StatusBubbleSettings from './StatusBubbleSettings'
import { USER_STATUS_AVAILABLE, USER_STATUS_BUSY, USER_STATUS_AWAY, USER_STATUS_OFFLINE } from '../../../constants'

const stories = storiesOf('Provider/v1/Atoms', module)
stories
  .addDecorator(withKnobs)
  .addDecorator(centered)
  .add('StatusBubbleSettings', () => {
    const statusOpts = {
      available: USER_STATUS_AVAILABLE,
      busy: USER_STATUS_BUSY,
      away: USER_STATUS_AWAY,
      offline: USER_STATUS_OFFLINE,
    }
    const statusVal = select('status', statusOpts, USER_STATUS_AVAILABLE)

    return <StatusBubbleSettings status={statusVal} />
  })
