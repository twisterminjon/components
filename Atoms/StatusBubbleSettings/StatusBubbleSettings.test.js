import React from 'react'

import StatusBubbleSettings from './StatusBubbleSettings'
import { USER_STATUS_AVAILABLE, USER_STATUS_BUSY, USER_STATUS_AWAY, USER_STATUS_OFFLINE } from '../../../constants'

describe('StatusBubbleSettings', () => {
  it('matches the snapshot for available', () => {
    const wrapper = window.shallow(<StatusBubbleSettings status={USER_STATUS_AVAILABLE} />)
    expect(wrapper).toMatchSnapshot()
  })

  it('matches the snapshot for busy', () => {
    const wrapper = window.shallow(<StatusBubbleSettings status={USER_STATUS_BUSY} />)
    expect(wrapper).toMatchSnapshot()
  })

  it('matches the snapshot for away', () => {
    const wrapper = window.shallow(<StatusBubbleSettings status={USER_STATUS_AWAY} />)
    expect(wrapper).toMatchSnapshot()
  })

  it('matches the snapshot for offline', () => {
    const wrapper = window.shallow(<StatusBubbleSettings status={USER_STATUS_OFFLINE} />)
    expect(wrapper).toMatchSnapshot()
  })
})
