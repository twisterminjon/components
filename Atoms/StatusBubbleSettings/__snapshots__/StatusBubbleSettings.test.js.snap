// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StatusBubbleSettings matches the snapshot for available 1`] = `
<div
  className="statusbubblesettings statusbubble--available"
  style={Object {}}
>
  <IconHamburger
    className=""
    color="var(--white)"
    size={9}
    style={Object {}}
  />
</div>
`;

exports[`StatusBubbleSettings matches the snapshot for away 1`] = `
<div
  className="statusbubblesettings statusbubble--away"
  style={Object {}}
>
  <IconHamburger
    className=""
    color="var(--white)"
    size={9}
    style={Object {}}
  />
</div>
`;

exports[`StatusBubbleSettings matches the snapshot for busy 1`] = `
<div
  className="statusbubblesettings statusbubble--busy"
  style={Object {}}
>
  <IconHamburger
    className=""
    color="var(--white)"
    size={9}
    style={Object {}}
  />
</div>
`;

exports[`StatusBubbleSettings matches the snapshot for offline 1`] = `
<div
  className="statusbubblesettings statusbubble--offline"
  style={Object {}}
>
  <IconHamburger
    className=""
    color="var(--white)"
    size={9}
    style={Object {}}
  />
</div>
`;
